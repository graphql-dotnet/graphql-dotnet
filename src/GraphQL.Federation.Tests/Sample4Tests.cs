using GraphQL.Federation.TypeFirst.Sample4;
using Microsoft.AspNetCore.Mvc.Testing;

namespace GraphQL.Federation.Tests;

public class Sample4Tests
{
    [Fact]
    public async Task Schema()
    {
        await using var webApp = new WebApplicationFactory<Program>();
        var ret = await webApp.Server.ExecuteGraphQLRequest("/graphql", "{ _service { sdl } }");
        ret.GetSdl().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public async Task Users()
    {
        await using var webApp = new WebApplicationFactory<Program>();
        var ret = await webApp.Server.ExecuteGraphQLRequest("/graphql", "{ users { id username } }");
        ret.ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public async Task Entities()
    {
        await using var webApp = new WebApplicationFactory<Program>();
        var query = "query ($arg: [_Any!]!) { _entities(representations: $arg) { __typename ... on User { id username } } }";
        var variables = """{ "arg": [{ "__typename": "User", "id": "1" },{ "__typename": "User", "id": "2" }] }""";
        var cat = await webApp.Server.ExecuteGraphQLRequest("/graphql", query, variables);
        cat.ShouldMatchApproved(o => o.NoDiff());
    }
}
