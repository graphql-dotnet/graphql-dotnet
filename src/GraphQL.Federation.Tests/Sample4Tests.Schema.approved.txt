schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@shareable", "@inaccessible", "@override", "@external", "@provides", "@requires", {name: "FieldSet", as: "fedFieldSet"}]) {
  query: Query
}

directive @link(url: String!, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA

directive @key(fields: String!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE

directive @shareable on FIELD_DEFINITION | OBJECT

directive @inaccessible on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

directive @override(from: String!) on FIELD_DEFINITION

directive @external on FIELD_DEFINITION | OBJECT

directive @provides(fields: federation__FieldSet!) on FIELD_DEFINITION

directive @requires(fields: federation__FieldSet!) on FIELD_DEFINITION

scalar _Any

type Query {
  users: [User!]!
  _service: _Service!
  _entities(representations: [_Any!]!): [_Entity]!
}

type User @key(fields: "id") {
  id: ID!
  username: String!
}

type _Service {
  sdl: String
}

union _Entity = User

enum link__Purpose {
  "`SECURITY` features provide metadata necessary to securely resolve fields."
  SECURITY
  "`EXECUTION` features provide metadata necessary for operation execution."
  EXECUTION
}

scalar link__Import

scalar federation__FieldSet